cannot call constructor 'PingUltrasonic::Thread'

Boa noite Ivan, 

Algumas classes de sensores que você desenvolveu contém o seguinte pedaço de código que está acusando o erro descrito no título:

Thread::Thread();

Na documentação do seu repositorio de Threads não encontrei referência a esse comando, então queria fazer o entendimento. Estou fazendo um robôzinho, e preciso que seja lido de forma "paralela" os valores do sensor ultrassônico HC-SR04.

Olá, estou enviando o log de erros.

Arduino: 1.6.8 (Windows 10), Placa:"Arduino/Genuino Uno"

```
C:\Program Files (x86)\Arduino\arduino-builder -dump-prefs -logger=machine -hardware "C:\Program Files (x86)\Arduino\hardware" -hardware "C:\Users\Jucemar\AppData\Local\Arduino15\packages" -tools "C:\Program Files (x86)\Arduino\tools-builder" -tools "C:\Program Files (x86)\Arduino\hardware\tools\avr" -tools "C:\Users\Jucemar\AppData\Local\Arduino15\packages" -built-in-libraries "C:\Program Files (x86)\Arduino\libraries" -libraries "E:\Arduino\Projetos\libraries" -fqbn=arduino:avr:uno -ide-version=10608 -build-path "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp" -warnings=all -prefs=build.warn_data_percentage=75 -verbose "C:\Users\Jucemar\Desktop\sketch_jun29a\sketch_jun29a.ino"
C:\Program Files (x86)\Arduino\arduino-builder -compile -logger=machine -hardware "C:\Program Files (x86)\Arduino\hardware" -hardware "C:\Users\Jucemar\AppData\Local\Arduino15\packages" -tools "C:\Program Files (x86)\Arduino\tools-builder" -tools "C:\Program Files (x86)\Arduino\hardware\tools\avr" -tools "C:\Users\Jucemar\AppData\Local\Arduino15\packages" -built-in-libraries "C:\Program Files (x86)\Arduino\libraries" -libraries "E:\Arduino\Projetos\libraries" -fqbn=arduino:avr:uno -ide-version=10608 -build-path "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp" -warnings=all -prefs=build.warn_data_percentage=75 -verbose "C:\Users\Jucemar\Desktop\sketch_jun29a\sketch_jun29a.ino"
"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10608 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\cores\arduino" "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\variants\standard" "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp\sketch\sketch_jun29a.ino.cpp" -o "nul"
"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10608 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\cores\arduino" "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\variants\standard" "-IE:\Arduino\Projetos\libraries\ArduinoThread" "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp\sketch\sketch_jun29a.ino.cpp" -o "nul"
"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10608 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\cores\arduino" "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\variants\standard" "-IE:\Arduino\Projetos\libraries\ArduinoThread" "-IE:\Arduino\Projetos\libraries\ArduinoSensors" "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp\sketch\sketch_jun29a.ino.cpp" -o "nul"
"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10608 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\cores\arduino" "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\variants\standard" "-IE:\Arduino\Projetos\libraries\ArduinoSensors" "-IE:\Arduino\Projetos\libraries\ArduinoThread" "E:\Arduino\Projetos\libraries\ArduinoThread\Thread.cpp" -o "nul"
"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10608 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\cores\arduino" "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\variants\standard" "-IE:\Arduino\Projetos\libraries\ArduinoSensors" "-IE:\Arduino\Projetos\libraries\ArduinoThread" "E:\Arduino\Projetos\libraries\ArduinoThread\ThreadController.cpp" -o "nul"
"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10608 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\cores\arduino" "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\variants\standard" "-IE:\Arduino\Projetos\libraries\ArduinoThread" "-IE:\Arduino\Projetos\libraries\ArduinoSensors" "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp\sketch\sketch_jun29a.ino.cpp" -o "nul"
"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avr-g++" -c -g -Os -w -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics  -w -x c++ -E -CC -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10608 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\cores\arduino" "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\variants\standard" "-IE:\Arduino\Projetos\libraries\ArduinoThread" "-IE:\Arduino\Projetos\libraries\ArduinoSensors" "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp\sketch\sketch_jun29a.ino.cpp" -o "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp\preproc\ctags_target_for_gcc_minus_e.cpp"
"C:\Program Files (x86)\Arduino\tools-builder\ctags\5.8-arduino10/ctags" -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp\preproc\ctags_target_for_gcc_minus_e.cpp"
"C:\Program Files (x86)\Arduino\hardware\tools\avr/bin/avr-g++" -c -g -Os -Wall -Wextra -std=gnu++11 -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -MMD -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=10608 -DARDUINO_AVR_UNO -DARDUINO_ARCH_AVR   "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\cores\arduino" "-IC:\Users\Jucemar\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.6.11\variants\standard" "-IE:\Arduino\Projetos\libraries\ArduinoThread" "-IE:\Arduino\Projetos\libraries\ArduinoSensors" "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp\sketch\sketch_jun29a.ino.cpp" -o "C:\Users\Jucemar\AppData\Local\Temp\build8739c9e26360eb8cf454482ee33346f7.tmp\sketch\sketch_jun29a.ino.cpp.o"
In file included from C:\Users\Jucemar\Desktop\sketch_jun29a\sketch_jun29a.ino:2:0:

E:\Arduino\Projetos\libraries\ArduinoSensors/ArduinoSensors.h:137:4: warning: #warning "ADC_DEFAULT_AREF = 5.0v | AVR >8Mhz & <=16Mhz clock" [-Wcpp]

   #warning "ADC_DEFAULT_AREF = 5.0v | AVR >8Mhz & <=16Mhz clock"

    ^

In file included from C:\Users\Jucemar\Desktop\sketch_jun29a\sketch_jun29a.ino:3:0:

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h: In constructor 'PingUltrasonic::PingUltrasonic(int, int)':

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h:86:18: error: cannot call constructor 'PingUltrasonic::Thread' directly [-fpermissive]

   Thread::Thread();

                  ^

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h:86:18: error:   for a function-style cast, remove the redundant '::Thread' [-fpermissive]

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h:86:18: error: no matching function for call to 'PingUltrasonic::PingUltrasonic()'

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h:86:18: note: candidates are:

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h:69:2: note: PingUltrasonic::PingUltrasonic(int, int)

  PingUltrasonic(int _trigPin, int _echoPin = -1){

  ^

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h:69:2: note:   candidate expects 2 arguments, 0 provided

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h:27:7: note: constexpr PingUltrasonic::PingUltrasonic(const PingUltrasonic&)

 class PingUltrasonic: public Thread, public DistanceInterface

       ^

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h:27:7: note:   candidate expects 1 argument, 0 provided

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h:27:7: note: constexpr PingUltrasonic::PingUltrasonic(PingUltrasonic&&)

E:\Arduino\Projetos\libraries\ArduinoSensors/sensors/PingUltrasonic.h:27:7: note:   candidate expects 1 argument, 0 provided

Usando a biblioteca ArduinoThread na vers�o 2.0.3 na pasta: E:\Arduino\Projetos\libraries\ArduinoThread 
Usando a biblioteca ArduinoSensors na vers�o 2.0 na pasta: E:\Arduino\Projetos\libraries\ArduinoSensors 
exit status 1
Error compiling for board Arduino/Genuino Uno.
```

--- a/sensors/PingUltrasonic.h
+++ b/sensors/PingUltrasonic.h
@@ -66,7 +66,7 @@ private:
 	}
 
 public:
-	PingUltrasonic(int _trigPin, int _echoPin = -1){
+	PingUltrasonic(int _trigPin, int _echoPin = -1): Thread() {
 		trigPin = _trigPin;
 
 		/*
@@ -83,7 +83,6 @@ public:
 		minDistance = 3;
 		maxDistance = 500;
 
-		Thread::Thread();
 		setInterval(50);
 	}
 

